packages/sdk/src/evm/common/legacy.ts

```diff
@@ -0,0 +1,55 @@
+import { ContractWrapper } from "../core/classes/contract-wrapper";
+import { hasFunction } from "./feature-detection";
+import type { IThirdwebContract } from "@thirdweb-dev/contracts-js";
+import IThirdwebContractABI from "@thirdweb-dev/contracts-js/dist/abis/IThirdwebContract.json";
+import { ethers } from "ethers";
+
+// TODO (cc) use this everywhere
+export async function isPrebuilt(
+  contractWrapper: ContractWrapper<any>,
+  contractType: string,
+  maxVersion: number,
+): Promise<boolean> {
+  if (hasFunction<IThirdwebContract>("contractType", contractWrapper)) {
+    try {
+      const [type, version] = await Promise.all([
+        ethers.utils.toUtf8String(
+          await contractWrapper.readContract.contractType(),
+        ),
+        await contractWrapper.readContract.contractVersion(),
+      ]);
+      const nameMatch = type.includes(contractType);
+      const versionMatch = maxVersion ? maxVersion <= version : true;
```

Comment:
Little edge case, if you put 0 as `maxVersion` for this function the version match will still return true. Also we would probably want `maxVersion` to be a `?` type in the function signature if we're checking for undefined here

Reply:
yeah good catch

Reply:
but ended up removing this function entirely